agent/grpc-internal/services/subscribe/subscribe.go

```diff
@@ -77,6 +77,15 @@ func (h *Server) Subscribe(req *pbsubscribe.SubscribeRequest, serverStream pbsub
 		case errors.Is(err, stream.ErrSubForceClosed):
 			logger.Trace("subscription reset by server")
 			return status.Error(codes.Aborted, err.Error())
+		case errors.Is(err, stream.ErrACLChanged):
+			logger.Trace("ACL change occurred; re-authenticating")
+			var authzErr error
+			authz, authzErr = h.Backend.ResolveTokenAndDefaultMeta(req.Token, &entMeta, nil)
```

Comment:
given that you don't use `authz` between here and returning, you could instead make this line be:
```
_, authzErr := h.Backend.ETCETC
```

Reply:
great call!